java/src/main/java/com/saucelabs/simplesauce/SauceOptions.java

```diff
@@ -89,130 +152,104 @@ public String getBuild() {
-
-    public SauceOptions withIE(String version) {
-        browserName = "IE";
-        browserVersion = version;
-        return this;
+    private void setPrimaryEnumCapability(String key, String value) {
+        switch (key) {
+            case "browserName":
+                if (!Options.Browser.keys().contains(value)) {
+                    String message = value + " is not a valid Browser, please choose from: " + Options.Browser.keys();
+                    throw new InvalidSauceOptionsArguementException(message);
+                } else {
+                    this.browserName = Options.Browser.valueOf(Options.Browser.fromString(value));
+                }
+                break;
+            case "platformName":
+                if (!Options.Platform.keys().contains(value)) {
+                    String message = value + " is not a valid Platform, please choose from: " + Options.Platform.keys();
+                    throw new InvalidSauceOptionsArguementException(message);
+                } else {
+                    this.platformName = Options.Platform.valueOf(Options.Platform.fromString(value));
+                }
+                break;
+            case "jobVisibility":
+                if (!Options.JobVisibility.keys().contains(value)) {
+                    String message = value + " is not a valid Job Visibility, please choose from: " + Options.JobVisibility.keys();
+                    throw new InvalidSauceOptionsArguementException(message);
+                } else {
+                    this.jobVisibility = Options.JobVisibility.valueOf(Options.JobVisibility.fromString(value));
+                }
+                break;
+            case "pageLoadStrategy":
+                if (!Options.PageLoadStrategy.keys().contains(value)) {
+                    String message = value + " is not a valid Job Visibility, please choose from: " + Options.PageLoadStrategy.keys();
+                    throw new InvalidSauceOptionsArguementException(message);
+                } else {
+                    this.pageLoadStrategy = Options.PageLoadStrategy.valueOf(Options.PageLoadStrategy.fromString(value));
+                }
+                break;
+            case "unhandledPromptBehavior":
+                if (!Options.UnhandledPromptBehavior.keys().contains(value)) {
+                    String message = value + " is not a valid Job Visibility, please choose from: " + Options.UnhandledPromptBehavior.keys();
+                    throw new InvalidSauceOptionsArguementException(message);
+                } else {
+                    this.unhandledPromptBehavior = Options.UnhandledPromptBehavior.valueOf(Options.UnhandledPromptBehavior.fromString(value));
+                }
+                break;
+        }
     }
 
-    public SauceOptions withIE() {
-        browserName = "IE";
-        browserVersion = "latest";
-        return this;
+    private void setSecondaryEnumCapability(String key, Map<String, Object> value) {
+        switch (key) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4758011)